src/components/ThePdfGenerator.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+  <div>
+    <embed :src="pdfUrl" type="application/pdf" width="100%" height="600px" />
+  </div>
+</template>
+
+<script>
+import jsPDF from "jspdf";
+import { useAppStore } from "@/stores/app";
+import { mapState } from "pinia";
+import autoTable from "jspdf-autotable";
+import { useDataStore } from "@/stores/data";
+import { avrileSansRegularNormal } from "@/assets/AvrileSans-Regular-normal.js";
+
+export default {
+  data() {
+    return {
+      pdfUrl: null,
+    };
+  },
+
+  computed: {
+    ...mapState(useAppStore, ["project"]),
+    ...mapState(useDataStore, [
+      "checkedTrenchesNames",
+      "checkedFieldNames",
+      "selectedType",
+      "projectPreferencesFieldsWithTranslation",
+      "checkedTrenchesItemsSelectedTypeAndSearched",
+    ]),
+    headers() {
+      return this.checkedFieldNames.map(
+        (fieldName) => this.projectPreferencesFieldsWithTranslation[fieldName]
+      );
+    },
+
+    rows() {
+      // select only fields present in checkedFieldNames
+      let arr = this.checkedFieldNames;
+      return this.checkedTrenchesItemsSelectedTypeAndSearched.map(function (
```

Comment:
@alzorba Ici tu peux utiliser une fat arrow, et donc ne pas avoir besoin de  `let arr = this.checkedFieldNames;` (qui sinon pourrait être un const, par ailleurs)`

Reply:
oui